pkg/segment/query/processor/inputlookupcommand_test.go

```diff
@@ -0,0 +1,306 @@
+		Start:          1,
+		Max:            3,
+	},
+	)
+
+	iqr, err := dp.Fetch()
+	assert.Nil(t, err)
+	assert.NotNil(t, iqr)
+	assert.Equal(t, 2, iqr.NumberOfRecords())
+
+	col_a, err := iqr.ReadColumn("a")
+	assert.Nil(t, err)
+	expected := []utils.CValueEnclosure{
+		{Dtype: utils.SS_DT_STRING, CVal: "a4"},
+		{Dtype: utils.SS_DT_STRING, CVal: "a7"},
+	}
+	assert.Equal(t, expected, col_a)
+
+	col_b, err := iqr.ReadColumn("b")
+	assert.Nil(t, err)
+	expected = []utils.CValueEnclosure{
+		{Dtype: utils.SS_DT_STRING, CVal: "b5"},
+		{Dtype: utils.SS_DT_STRING, CVal: "b8"},
+	}
+	assert.Equal(t, expected, col_b)
+
+	col_c, err := iqr.ReadColumn("c")
+	assert.Nil(t, err)
+	expected = []utils.CValueEnclosure{
+		{Dtype: utils.SS_DT_STRING, CVal: "c6"},
+		{Dtype: utils.SS_DT_STRING, CVal: "c9"},
+	}
+	assert.Equal(t, expected, col_c)
+
+	iqr, err = dp.Fetch()
+	assert.Equal(t, io.EOF, err)
+	assert.Nil(t, iqr)
+
+	os.RemoveAll(config.GetDataPath())
+}
+
+func getExpectedColumnValues(val string, start int, count int) []utils.CValueEnclosure {
+	var expected []utils.CValueEnclosure
+	for i := 0; i < count; i++ {
+		colValue := fmt.Sprintf("%v%v", val, start+i)
+		expected = append(expected, utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: colValue})
+	}
+	return expected
+}
+
+func validate_MultiFetch(t *testing.T, dp *DataProcessor, expectedCols map[string]struct{}, start []int, numRecords []int, numFetches int) {
+	fetch := 0
+	for {
+		iqr, err := dp.Fetch()
+		if fetch == numFetches {
+			assert.Equal(t, io.EOF, err)
+			break
+		} else {
+			assert.Nil(t, err)
+		}
+		assert.NotNil(t, iqr)
+		assert.Equal(t, numRecords[fetch], iqr.NumberOfRecords())
+		actualCols, err := iqr.GetColumns()
+		assert.Nil(t, err)
+		assert.Equal(t, expectedCols, actualCols)
+
+		// Column values from 2 to 101 would be read
+		for col := range expectedCols {
+			colValues, err := iqr.ReadColumn(col)
+			assert.Nil(t, err)
+			assert.Equal(t, getExpectedColumnValues(col, start[fetch], numRecords[fetch]), colValues)
+		}
+		fetch++
+	}
+
+	assert.Equal(t, numFetches, fetch)
+}
+
+func Test_InputLookupMultipleFetchAndRewind(t *testing.T) {
+
+	err := initTestConfig(t)
+	assert.Nil(t, err)
+
+	data := []string{"a,b,c\n"}
+	for i := 0; i < 110; i++ {
+		idx := fmt.Sprintf("%v", i)
+		row := "a" + idx + ",b" + idx + ",c" + idx + "\n"
+		data = append(data, row)
+	}
+
+	csvFile := "test.csv"
+
+	err = prepareData(data, csvFile)
+	assert.Nil(t, err)
+
+	dp := NewInputLookupDP(&structs.InputLookup{
+		IsFirstCommand: true,
+		Filename:       csvFile,
+		Start:          2,
+		Max:            10000,
+	},
+	)
+
+	expectedCols := map[string]struct{}{
+		"a": {},
+		"b": {},
+		"c": {},
+	}
+	start := []int{2, 102}
+	numRecords := []int{100, 8}
+
+	validate_MultiFetch(t, dp, expectedCols, start, numRecords, 2)
+
+	// Rewind to the start
+	dp.Rewind()
+
+	validate_MultiFetch(t, dp, expectedCols, start, numRecords, 2)
+
+	os.RemoveAll(config.GetDataPath())
```

Comment:
nit: It's better to defer this in a `t.Cleanup()` at the start of the test; if the test panics, it will not reach but the functions in `t.Cleanup()` will still get called

See https://pkg.go.dev/testing#T.Cleanup

Reply:
Just using the tempDir, so need of this. addressed in the latest commit.